File: replace.py
import os

dir_path = '/home/dustin/Projects/CommandCenter'
files_to_process = ['index.html', 'script.js']

replacements = [
    # General UI / Text
    ("PromptForge", "CommandCenter"),
    ("promptForge", "commandCenter"),
    ("AI Prompt Lab", "AI Command Lab"),
    ("AI Prompt Management System", "AI Command Management System"),
    ("AI Enhancement Settings", "AI Recommendation Settings"),
    ("Enhancement Prompt", "Recommendation Prompt"),
    ("Enter system prompt for enhancement...", "Enter system prompt for recommendation..."),
    ("Describe the prompt you want to create or enhance...", "Describe the command you want to create or fix..."),
    ("New Prompt", "New Command"),
    ("Search prompts...", "Search commands..."),
    ("No Prompts Found", "No Commands Found"),
    ("no prompts matching", "no commands matching"),
    ("Prompt Details", "Command Details"),
    ("Copy Prompt", "Copy Command"),
    ("Export All Prompts", "Export All Commands"),
    ("Keeps your current prompts", "Keeps your current commands"),
    ("prompt content to enhance", "command content to fix"),
    ("Prompt Content", "Command"),
    ("e.g., Creative Writing Assistant", "e.g., Start Node.js server"),
    ("e.g., You are a creative writing assistant...", "e.g., node server.js"),
    ("fa-hammer", "fa-terminal"),
    # AI System prompt
    ("You are an expert prompt engineer. Improve the prompt for clarity and effectiveness while preserving intent. Return only the improved prompt text.", "You are an expert Linux sysadmin. Suggest or fix the command described by the user to achieve their goal. Return only the command text, without markdown formatting or code blocks."),
    # JS UI strings
    ("Enhancing...", "Processing..."),
    ("Enhance with AI", "Suggest/Fix Command"),
    ("Prompt enhanced successfully!", "Command fixed successfully!"),
    ("Enhance prompt", "Suggest/Fix command"),
    ("No prompt content to enhance", "No command content to suggest/fix")
]

for filename in files_to_process:
    filepath = os.path.join(dir_path, filename)
    if os.path.exists(filepath):
        with open(filepath, 'r') as f:
            content = f.read()
        
        new_content = content
        for old, new in replacements:
            new_content = new_content.replace(old, new)
        
        if content != new_content:
            with open(filepath, 'w') as f:
                f.write(new_content)
            print(f"Updated {filename}")

File: replace.js
const fs = require('fs');
const path = require('path');

const dir = '/home/dustin/Projects/CommandCenter';

function replaceInFile(filePath, searchRegex, replacement) {
    if (fs.existsSync(filePath)) {
        let content = fs.readFileSync(filePath, 'utf8');
        let newContent = content.replace(searchRegex, replacement);
        if (content !== newContent) {
            fs.writeFileSync(filePath, newContent);
            console.log(`Updated ${filePath}`);
        }
    }
}

function globalReplace(searchStr, replacement) {
    const files = ['index.html', 'script.js', 'style.css', 'manifest.json', 'README.md', 'offline.html', 'importExport.js'];
    files.forEach(f => {
        replaceInFile(path.join(dir, f), new RegExp(searchStr, 'g'), replacement);
    });
}

// 1. PromptForge -> CommandCenter
globalReplace('PromptForge', 'CommandCenter');
globalReplace('promptForge', 'commandCenter');

// 2. hammer icon to terminal
replaceInFile(path.join(dir, 'index.html'), /fa-hammer/g, 'fa-terminal');

// 3. AI Prompts -> Commands
replaceInFile(path.join(dir, 'script.js'), /You are an expert prompt engineer.*?Return only the improved prompt text\./g, 'You are an expert Linux sysadmin. Suggest or fix the command described by the user to achieve their goal. Return only the command text, without markdown formatting or code blocks.');

// Update UI text in script.js
replaceInFile(path.join(dir, 'script.js'), /<i class="fas fa-spinner fa-spin"><\/i> Enhancing\.\.\./g, '<i class="fas fa-spinner fa-spin"></i> Processing...');
replaceInFile(path.join(dir, 'script.js'), /<i class="fas fa-magic"><\/i> Enhance with AI/g, '<i class="fas fa-magic"></i> Suggest/Fix Command');
replaceInFile(path.join(dir, 'script.js'), /Prompt enhanced successfully!/g, 'Command processed successfully!');
replaceInFile(path.join(dir, 'script.js'), /No prompt content to enhance/g, 'No command content to suggest/fix');
replaceInFile(path.join(dir, 'script.js'), /Enhancing\.\.\./g, 'Processing...');
replaceInFile(path.join(dir, 'script.js'), /Enhance prompt/gi, 'Suggest/Fix command');

// Update UI text in index.html
replaceInFile(path.join(dir, 'index.html'), /AI Prompt Lab/g, 'AI Command Lab');
replaceInFile(path.join(dir, 'index.html'), /AI Prompt Management System/g, 'AI Command Management System');
replaceInFile(path.join(dir, 'index.html'), /Describe the prompt you want to create or enhance/g, 'Describe the command you want to create or fix');

console.log('Done.');
